File: src/main/scala/category/greedy/_452_Solution.scala
package category.greedy

/**
  * 452. Minimum Number of Arrows to Burst Balloons
  */
object _452_Solution {
  def findMinArrowShots(points: Array[Array[Int]]): Int = {
    if (points.isEmpty || points(0).isEmpty) return 0
    val sorted = points.sortWith { case (_@Array(_, s1), _@Array(_, t1)) => s1 < t1 }
    var end = sorted(0).last
    var count = 1
    for (i <- 1 until sorted.length) {
      if (sorted(i).head > end) {count += 1; end = sorted(i).last}
    }
    count
  }
}

File: src/main/python/medium/_450_500/_452_Solution.py
# 452. Minimum Number of Arrows to Burst Balloons
from typing import List


class _452_Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points or not points[0]: return 0
        n, ans = len(points), 0
        end = -0xfffffff
        points.sort(key=lambda lst: lst[1])
        for point in points:
            if point[0] > end:
                end = points[1]
                ans += 1
        return ans
